File: Challenges/Ruby/excelSheetColumnNumber.rb
def excelSheetColumnNumber(s)
	sum, l = 0, s.size - 1
	s.size.times do |i|
		a = s[i].ord - 64
		l.times { a *= 26; }
		sum, l = sum + a, l - 1
	end
	return sum
end

File: Challenges/Java/ExcelSheetColumnNumber.java
public final class ExcelSheetColumnNumber {

	int excelSheetColumnNumber(String s) {
	    int sum = 0, l = s.length() - 1;
	    for(int i = 0; i < s.length(); i++, l--) {
	        int a = (byte)s.charAt(i) - 64;
	        for(int j = 0; j < l; j++) { a *= 26; }
	        sum += a;
	    }
	    return sum;
	}
}